src/averageAges.js

```diff
@@ -1,61 +1,51 @@
 'use strict';
 
-/**
- * Implement calculateMenAverageAge function
- *
- * Function returns average age of men in array. If `century` is specified then
- * function calculates average age only for men who died in this century
- *
- * To calculate century:
- * Divide year of person's death by 100: Math.ceil(person.died / 100)
- *
- * @param {object[]} people
- * @param {number} century - optional
- *
- * @return {number}
- */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  let manArray = [];
+
+  century
+    ? manArray = people.filter((man) => man.sex === 'm'
+    && Math.ceil(man.died / 100) === century)
+
+    : manArray = people.filter((man) => man.sex === 'm');
+
+  return manArray.reduce(
+    (averageAge, person) => averageAge + (person.died - person.born),
+    0,
+  ) / manArray.length;
```

Comment:
you do the same on calculateWomenAverageAge. You can create separate function and you it here and on calculateWomenAverageAge